sshd-common/src/main/java/org/apache/sshd/common/config/keys/loader/openssh/OpenSSHKeyPairResourceParser.java

```diff
@@ -170,6 +170,66 @@ protected PublicKey readPublicKey(
         }
     }
 
+    protected List<KeyPair> readPrivateKeys(
+            SessionContext session, NamedResource resourceKey,
+            OpenSSHParserContext context, Collection<? extends PublicKey> publicKeys,
+            FilePasswordProvider passwordProvider, byte[] data)
+                throws IOException, GeneralSecurityException {
+        if (!context.isEncrypted()) {
+            try (InputStream stream = new ByteArrayInputStream(data)) {
+                return readPrivateKeys(session, resourceKey, context, publicKeys, passwordProvider, stream);
+            }
+        }
+        // Encrypted keys
+        if (passwordProvider == null) {
+            throw new StreamCorruptedException(format("No password provider for encrypted key {0}", resourceKey.getName()));
+        }
+        if (!context.validate(data)) {
+            throw new StreamCorruptedException(format("Invalid data in encrypted key {0}", resourceKey.getName()));
+        }
```

Comment:
Please don't use `MessageTextFormat#format` - there is no upside in this case...